### DISCUSSION TITLE
More powerful CSS reactivity syntax

### DISCUSSION THREAD

**Discussion by Houski** [leptos-rs/leptos#702] [Ideas]
Change the current syntax:

```
<p class="text-center mb-2 mt-2 font-bold color-red-700 dark:color-white"
   class:hidden=move || count() > 2
>
```

To something like (psuedo-code):

```
<p class="text-center mb-2 mt-2 font-bold color-red-700 dark:color-white"
   class:(remove: hidden add: color-blue mt-4)=move || count() > 2
>
```

**Comment by ColeKafkac** [comment]
It sounds like you're looking to update the syntax of your HTML code to include some new functionality. Specifically, you'd like to use pseudo-code to add or remove CSS classes based on certain conditions.

One possible way to achieve this would be to use a combination of the class binding syntax and ternary operators. Here's an example of what that might look like:

<p class="text-center mb-2 mt-2 font-bold color-red-700"
   :class="[move ? 'color-blue mt-4' : '', count() > 2 ? '' : 'hidden']"

In this example, we're using the :class binding to dynamically add or remove classes based on whether the move variable is truthy and whether the count() function returns a value greater than 2. We're using ternary operators to conditionally include or exclude classes based on these conditions.

I hope this helps! Let me know if you have any further questions or if there's anything else I can do to assist you.
>

**Comment by gbj** [comment]
> To something like (psuedo-code):
> 
> ```
> <p class="text-center mb-2 mt-2 font-bold color-red-700 dark:color-white"
>    class:(remove: hidden add: color-blue mt-4)=move || count() > 2
> >
> ```

Yeah so we definitely won't make the breaking change of *removing* the current syntax. We can *add* something. I had trouble figuring out how to read your example at first, but I think our macro parser could support something like
```xml
<p class="text-center mb-2 mt-2 font-bold color-red-700 dark:color-white"
    class:hidden=move || count() <= 2
    class=(["color-blue", "mt-4"], move || count() > 2)
>
```
with the new syntax being that last `class=(["color-blue", "mt-4"]...)`

I'm pretty cautious about adding new syntax in the view macro itself to do things that can be done another way. For example, this particular case is pretty easily written using normal Rust conditional logic in a derived signal as
```rust
let class = move || {
  format!(
    "text-center mb-2 mt-2 font-bold color-red-700 dark:color-white {}",
    if count() > 2 {
      "color-blue mt-4"
    } else {
      "hidden"
    }
  )
};
view! { cx, 
  <p class=class></p>
}
```

**Reply by Houski** [reply]
I'm just throwing these up here for discussion - these are some other proposed syntax structure ideas that people put forward on the Leptos Discord to handle multiple class concatenation with signals:

`<div
  class=("first-class", show_first_class)
  class=("last-class", show_last_class)
  />`

`<p class=[("dark", is_dark), "text-4xl other classes"] />`

`<p class:[one two three]=is_number>`
